File: ide-features-trainer/src/training/commands/TextCommand.java
package training.commands;

import org.jdom.Element;
import training.learn.Lesson;

/**
 * Created by karashevich on 30/01/15.
 */
public class TextCommand extends Command {

    public TextCommand(){
        super(CommandType.TEXT);
    }

    @Override
    public void execute(ExecutionList executionList) throws InterruptedException {


        Element element = executionList.getElements().poll();
        Lesson lesson = executionList.getLesson();

        String htmlText = (element.getContent().isEmpty() ? "" : element.getContent().get(0).getValue());
        if (htmlText.isEmpty()) htmlText = element.getAttribute("description").getValue();

        if (htmlText.equals("")) {
            updateDescription(htmlText, lesson);
        } else {
            updateHTMLDescription(htmlText, lesson);
        }

        startNextCommand(executionList);

    }

}

File: ide-features-trainer/src/training/commands/TextCommand.kt
// Copyright 2000-2019 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE file.
package training.commands

import training.learn.lesson.LessonManager

class TextCommand : Command(CommandType.TEXT) {

  @Throws(InterruptedException::class)
  override fun execute(executionList: ExecutionList) {
    val element = executionList.elements.poll()
    LessonManager.instance.addMessages(element)
    startNextCommand(executionList)
  }

}
